src/python/oftest/controller.py

```diff
@@ -295,8 +295,6 @@ def _pkt_handle(self, pkt):
                         code_str = "unknown"
                     self.logger.warn("Received error message: xid=%d type=%s (%d) code=%s (%d)",
                                      hdr_xid, type_str, msg.err_type, code_str, msg.code if code_str != "unknown" else -1)
-                    if msg.version >= 3 and isinstance(msg, ofp.message.bsn_error):
-                        self.logger.warn("BSN error, msg '%s'", msg.err_msg)
```

Comment:
instead of nuking this, how about wrapping it inside of a try/except?

Reply:
Maybe the real problem is that bsn_error exists in OF1.3 and later but "msg.version >= 3" tests for OF1.2 and later.  I'll check and resubmit as a change to "msg.version >= 4" if so.